apps/api/Dockerfile

```diff
@@ -3,34 +3,33 @@ ENV PNPM_HOME="/pnpm"
 ENV PATH="$PNPM_HOME:$PATH"
 LABEL fly_launch_runtime="Node.js"
 RUN corepack enable
-COPY . /app
+
 WORKDIR /app
 
+# Install dependencies only
 FROM base AS prod-deps
+COPY package.json pnpm-lock.yaml ./
 RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --prod --frozen-lockfile
 
+# Install all dependencies and build the application
 FROM base AS build
-RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
-
-RUN pnpm install
-RUN pnpm run build
-
-# Install packages needed for deployment
-
-
-FROM base
-RUN apt-get update -qq && \
-    apt-get install --no-install-recommends -y chromium chromium-sandbox && \
-    rm -rf /var/lib/apt/lists /var/cache/apt/archives
-COPY --from=prod-deps /app/node_modules /app/node_modules
-COPY --from=build /app /app
+COPY . ./
```

Comment:
Copying the code here instead of line 6 like before, because, we don't need the whole source code to install all pnpm deps.